Autopilot/PathManager/Src/cruisingState.cpp

```diff
@@ -0,0 +1,349 @@
+using namespace std;
+
+ /*
+    Integer codes from the telemetryData object:
+
+        waypointModifyFlightPathCommand:
+            0 = nothing
+            1 = initialize flight path
+            2 = append waypoint
+            3 = insert waypoint
+            4 = update waypoint
+            5 = delete waypoint
+            6 = nuke flight path (remove all waypoints)
+
+        waypointNextDirectionsCommand:
+            0 = nothing (regular path following)
+            1 = start/end holding
+            2 = head home
+        
+        initializingHomeBase:
+            1 = yes
+            2 = no
+
+    Integer codes for error codes:
+        editingFlightPathErrorCode:
+            0 = success
+            1 = error
+            2 = incorrect telemetry command
+
+        pathFollowingErrorCode:
+            0 = success
+            1 = error
+            2 = incorrect telemetry command
+            3 = homebase not initialized
+*/
+
+
+int editFlightPath(Telemetry_PIGO_t * telemetryData, WaypointManager& cruisingStateManager, int * idArray) {
+    
+    // If no commands given, just skip over this function
+    if (telemetryData->waypointModifyFlightPathCommand == 0) {        
+        return 0;
+    }
+    
+    _WaypointOutputType waypointType = PATH_FOLLOW;
+    _WaypointStatus editingStatus = WAYPOINT_SUCCESS;
+
+    /* Editing the flight path */
+
+    if (telemetryData->numWaypoints == 1 && (telemetryData->waypointModifyFlightPathCommand >= 2 && telemetryData->waypointModifyFlightPathCommand <= 4)) { // Inserting, Appending, or Updating a waypoint
+        
+        // cout << "Editing flight path (Not deleting)" << endl;
+
+        // Set the output type of the new waypoint. Need if statements because of the enum parameter of the initialize_waypoint() method
+        if (telemetryData->waypoints[0]->waypointType == 0) {
+            waypointType = PATH_FOLLOW;
+        } else if (telemetryData->waypoints[0]->waypointType == 1) {
+            waypointType = ORBIT_FOLLOW;
+        } else {
+            waypointType = HOLD_WAYPOINT;
+        }
+
+        // Depending on the waypointType of the waypoint, we will need to call a different initialize_waypoint() method
+        _PathData * modifyWaypoint;
+        if (waypointType == PATH_FOLLOW) {
+            modifyWaypoint = cruisingStateManager.initialize_waypoint(telemetryData->waypoints[0]->longitude, telemetryData->waypoints[0]->latitude, telemetryData->waypoints[0]->altitude, waypointType); 
+        } else {
+            modifyWaypoint = cruisingStateManager.initialize_waypoint(telemetryData->waypoints[0]->longitude, telemetryData->waypoints[0]->latitude, telemetryData->waypoints[0]->altitude, waypointType, telemetryData->waypoints[0]->turnRadius); // Create a _PathData object
+        }
+        
+        // Update flight path by passing in the appropriate parameters to update_path_nodes()
+        if (telemetryData->waypointModifyFlightPathCommand == 2) { // Append
```

Comment:
A suggestion. Want to create an enum for these commands ? might make things a bit more readable.